src/buffer_manager.ts

```diff
@@ -410,6 +412,74 @@ export class BufferManager implements Disposable {
 
     private handleWindowChangedDebounced = debounce(this.handleWindowChanged, 100, { leading: false, trailing: true });
 
+    private handleSaveBuf(bufID: number, filename: string) {
+        const document = this.getTextDocumentForBufferId(bufID);
+        if (document == null) {
+            throw new Error("Received an invalid buffer id from Neovim, cannot save");
+        }
+
+        this.saveDocument(bufID, document, filename).catch((err) => {
+            window.showErrorMessage(`Failed to save document to ${filename}: ${err}`);
+        });
+    }
+
+    private async saveDocument(bufID: number, document: TextDocument, filename: string): Promise<void> {
+        const needSaveAs = await this.needSaveAs(document, filename);
+        if (needSaveAs) {
```

Comment:
The idea here is that because `saveas` is actually the complicated case, we're going to let VSCode do a standard `saveas`, and handle all of the editor renaming for us. This will have to change in the future if we want better integration, but as @theol0403 pointed out, this isn't a regression from current behavior (which always opens the dialog anyway)